main.js

```diff
@@ -1,46 +1,108 @@
-const { app, BrowserWindow, Tray, Menu } = require('electron');
-const path = require('path');
+import { app, Tray, Menu, dialog, shell } from "electron";
+import express from "express";
+import path from "path";
+import { fileURLToPath } from "url";
+import { SerialPort } from "serialport";
+import fs from "fs";
+import AdmZip from "adm-zip";
 
-let tray = null;
-let mainWindow = null;
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+let tray;
 
-app.whenReady().then(() => {
-    // Create system tray icon
-    tray = new Tray(path.join(__dirname, 'icon.png')); // Add a custom icon
-    const contextMenu = Menu.buildFromTemplate([
-        { label: 'Open ElectroBlocks', click: () => createWindow() },
-        { type: 'separator' },
-        { label: 'Quit', click: () => app.quit() }
-    ]);
-    tray.setToolTip('ElectroBlocks Tray App');
-    tray.setContextMenu(contextMenu);
-
-    // Create window when clicking tray icon
-    tray.on('click', () => {
-        if (mainWindow) {
-            mainWindow.show();
-        } else {
-            createWindow();
-        }
+const unzipBuildIfNeeded = () => {
+  const zipPath = path.join(__dirname, "build.zip");
+  const buildPath = path.join(__dirname, "build");
+
+  if (fs.existsSync(zipPath) && !fs.existsSync(buildPath)) {

```

Comment:
This works.  You unzipped the file into the build/build directory. Please confirm that you are testing this on your local machine. I am not trying to be mean, but I need everyone to QA their work.  

```js
const unzipBuildIfNeeded = () => {
  const zipPath = path.join(__dirname, "build.zip");
  const buildPath = path.join(__dirname, "build");

  if (fs.existsSync(zipPath) && !fs.existsSync(buildPath)) {
    try {
      const zip = new AdmZip(zipPath);
      zip.extractAllTo(__dirname, true);
      fs.unlinkSync(zipPath);
      console.log("Unzipped build.zip and deleted the zip file.");
    } catch (err) {
      console.error("Error unzipping build.zip:", err);
    }
  }
};
```